File: Core/src/com/stusherwin/midiapp/core/OutputNode.java
package com.stusherwin.midiapp.core;

import rx.Observer;

public class OutputNode<T> extends Node<T> {
    public void connectTo(InputNode<T> input) {
        input.connectTo(this);
    }

    public Observer<T> observer() {
        return this.subject;
    }
}

File: Core/src/com/stusherwin/midiapp/core/modules/OutputNode.scala
package com.stusherwin.midiapp.core.modules

import rx.lang.scala.Observer

class OutputNode[T] extends Node[T] {
  val observer : Observer[T] = this.subject

  def connectTo(input : InputNode[T]) : Unit = {
    input.connectTo(this)
  }
}